src/model/HeatPumpPlantLoopEIRCooling.cpp

```diff
@@ -0,0 +1,491 @@
+      : WaterToWaterComponent_Impl(idfObject, model, keepHandle) {
+      OS_ASSERT(idfObject.iddObject().type() == HeatPumpPlantLoopEIRCooling::iddObjectType());
+    }
+
+    HeatPumpPlantLoopEIRCooling_Impl::HeatPumpPlantLoopEIRCooling_Impl(const openstudio::detail::WorkspaceObject_Impl& other, Model_Impl* model,
+                                                                       bool keepHandle)
+      : WaterToWaterComponent_Impl(other, model, keepHandle) {
+      OS_ASSERT(other.iddObject().type() == HeatPumpPlantLoopEIRCooling::iddObjectType());
+    }
+
+    HeatPumpPlantLoopEIRCooling_Impl::HeatPumpPlantLoopEIRCooling_Impl(const HeatPumpPlantLoopEIRCooling_Impl& other, Model_Impl* model,
+                                                                       bool keepHandle)
+      : WaterToWaterComponent_Impl(other, model, keepHandle) {}
+
+    const std::vector<std::string>& HeatPumpPlantLoopEIRCooling_Impl::outputVariableNames() const {
+      static const std::vector<std::string> result{"Heat Pump Electricity Energy",
+                                                   "Heat Pump Load Side Heat Transfer Energy",
+                                                   "Heat Pump Source Side Heat Transfer Energy",
+                                                   "Heat Pump Electricity Rate",
+                                                   "Heat Pump Load Side Heat Transfer Rate",
+                                                   "Heat Pump Source Side Heat Transfer Rate",
+                                                   "Heat Pump Load Side Outlet Temperature",
+                                                   "Heat Pump Load Side Inlet Temperature",
+                                                   "Heat Pump Source Side Outlet Temperature",
+                                                   "Heat Pump Source Side Inlet Temperature",
+                                                   "Heat Pump Load Side Mass Flow Rate",
+                                                   "Heat Pump Source Side Mass Flow Rate"};
+      return result;
+    }
+
+    IddObjectType HeatPumpPlantLoopEIRCooling_Impl::iddObjectType() const {
+      return HeatPumpPlantLoopEIRCooling::iddObjectType();
+    }
+
+    unsigned HeatPumpPlantLoopEIRCooling_Impl::supplyInletPort() const {
+      return OS_HeatPump_PlantLoop_EIR_CoolingFields::SourceSideInletNodeName;
+    }
+
+    unsigned HeatPumpPlantLoopEIRCooling_Impl::supplyOutletPort() const {
+      return OS_HeatPump_PlantLoop_EIR_CoolingFields::SourceSideOutletNodeName;
+    }
+
+    unsigned HeatPumpPlantLoopEIRCooling_Impl::demandInletPort() const {
+      return OS_HeatPump_PlantLoop_EIR_CoolingFields::LoadSideInletNodeName;
+    }
+
+    unsigned HeatPumpPlantLoopEIRCooling_Impl::demandOutletPort() const {
+      return OS_HeatPump_PlantLoop_EIR_CoolingFields::LoadSideOutletNodeName;
+    }
```

Comment:
I fear this is backwards.

```suggestion
    unsigned HeatPumpPlantLoopEIRCooling_Impl::supplyInletPort() const {
      return OS_HeatPump_PlantLoop_EIR_CoolingFields::LoadSideInletNodeName;
    }

    unsigned HeatPumpPlantLoopEIRCooling_Impl::supplyOutletPort() const {
      return OS_HeatPump_PlantLoop_EIR_CoolingFields::LoadSideOutletNodeName;
    }

    unsigned HeatPumpPlantLoopEIRCooling_Impl::demandInletPort() const {
      return OS_HeatPump_PlantLoop_EIR_CoolingFields::SourceSideInletNodeName;
    }

    unsigned HeatPumpPlantLoopEIRCooling_Impl::demandOutletPort() const {
      return OS_HeatPump_PlantLoop_EIR_CoolingFields::SourceSideOutletNodeName;
    }
```


For context: **PlantLoopHeatPump_EIR_WaterSource.idf**

```
  HeatPump:PlantLoop:EIR:Heating,
    Heating Coil,            !- Name
    Heating Coil Load Loop Intermediate Node,  !- Load Side Inlet Node Name
    Heating Coil Load Loop Supply Outlet Node,  !- Load Side Outlet Node Name
    WaterSource,             !- Condenser Type
    Condenser Loop Demand Inlet Node,  !- Source Side Inlet Node Name
    Condenser Loop Demand Intermediate Node,  !- Source Side Outlet Node Name
```

https://github.com/NREL/OpenStudio/blob/2305735c38fbfa5e7ae9621558ae87a8d8b3f97d/src/model/ChillerElectricEIR.cpp#L564-L578